dynamic-simulation/dynamic-simulation-tool/src/main/java/com/powsybl/dynamicsimulation/tool/DynamicSimulationTool.java

```diff
@@ -118,22 +120,30 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
         }
 
         context.getOutputStream().println("Loading network '" + caseFile + "'");
-        Properties inputParams = readProperties(line, ConversionToolUtils.OptionType.IMPORT, context);
+        Properties inputParams = ConversionToolUtils.readProperties(line, ConversionToolUtils.OptionType.IMPORT, context);
         ImportConfig importConfig = (!skipPostProc) ? ImportConfig.load() : new ImportConfig();
         Network network = Importers.loadNetwork(caseFile, context.getShortTimeExecutionComputationManager(),
             importConfig, inputParams);
         if (network == null) {
             throw new PowsyblException("Case '" + caseFile + "' not found");
         }
 
+        DynamicSimulation.Runner runner = DynamicSimulation.find();
+
+        CurvesProvider curvesProvider = CurvesProvider.empty();
+        if (line.hasOption(CURVES_FILE)) {
+            Path curvesFile = context.getFileSystem().getPath(line.getOptionValue(CURVES_FILE));
+            curvesProvider = createCurvesProvider(curvesFile, runner.getName());
+        }
+
         DynamicSimulationParameters params = DynamicSimulationParameters.load();
         if (line.hasOption(PARAMETERS_FILE)) {
             Path parametersFile = context.getFileSystem().getPath(line.getOptionValue(PARAMETERS_FILE));
             JsonDynamicSimulationParameters.update(params, parametersFile);
         }
 
-        DynamicSimulationResult result = DynamicSimulation.run(network,
-            context.getShortTimeExecutionComputationManager(), params);
+        // TODO: use the curvesProvider
```

Comment:
We should fix this TODO before merging this PR. @mathbagu did you leave it for @marcosmc to complete?

Reply:
Done.